src/plugins/intel_cpu/src/nodes/sparse_fill_empty_rows.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "sparse_fill_empty_rows.h"
+
+#include "openvino/op/sparse_fill_empty_rows.hpp"
+#include "openvino/reference/sparse_fill_empty_rows.hpp"
+
+namespace ov::intel_cpu::node {
+SparseFillEmptyRows::SparseFillEmptyRows(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context)
+    : Node(op, context, NgraphShapeInferFactory(op)) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        OPENVINO_THROW_NOT_IMPLEMENTED(errorMessage);
+    }
+}
+
+bool SparseFillEmptyRows::isSupportedOperation(const std::shared_ptr<const ov::Node>& op,
+                                               std::string& errorMessage) noexcept {
+    try {
+        if (!ov::is_type<ov::op::v16::SparseFillEmptyRows>(op)) {
+            errorMessage = "Only opset16 SparseFillEmptyRows operation is supported";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+void SparseFillEmptyRows::getSupportedDescriptors() {
+    // Validation is already done in the ov::opset16::SparseFillEmptyRows
+}
+
+void SparseFillEmptyRows::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty()) {
+        return;
+    }
+    ov::element::Type valuesPrecision = getOriginalInputPrecisionAtPort(0);
+    ov::element::Type indicesPrecision = getOriginalInputPrecisionAtPort(2);
+    addSupportedPrimDesc({{LayoutType::ncsp, valuesPrecision},        // values
+                          {LayoutType::ncsp, indicesPrecision},       // dense_shape
+                          {LayoutType::ncsp, indicesPrecision},       // indices
+                          {LayoutType::ncsp, valuesPrecision}},       // default_value
+                         {{LayoutType::ncsp, indicesPrecision},       // output_indices
+                          {LayoutType::ncsp, valuesPrecision},        // output_values
+                          {LayoutType::ncsp, ov::element::boolean}},  // empty_row_indicator
+                         impl_desc_type::ref);
+}
+
+bool SparseFillEmptyRows::created() const {
+    return getType() == Type::SparseFillEmptyRows;
+}
+
+bool SparseFillEmptyRows::needPrepareParams() const {
+    return false;
+}
+
+bool SparseFillEmptyRows::isExecutable() const {
+    return !isInputTensorAtPortEmpty(1) && !isInputTensorAtPortEmpty(3);
+}
+
+void SparseFillEmptyRows::executeDynamicImpl(const dnnl::stream& strm) {
+    const auto& valuesMemory = getSrcMemoryAtPort(0);
+    const auto& indicesMemory = getSrcMemoryAtPort(2);
+    const auto& valuesShape = valuesMemory->getShape();
+    const auto& indicesShape = indicesMemory->getShape();
+    int64_t numRows = 0;
+
+    const auto* denseShapePtr = getSrcDataAtPortAs<const int32_t>(1);
+    numRows = static_cast<int64_t>(denseShapePtr[0]);
```

Comment:
```suggestion
    const auto numRows = static_cast<size_t>(denseShapePtr[0]);
```
can it be size_t? Some cast can be avoided then.

Reply:
Done.